core/src/main/scala/org/apache/spark/util/collection/BitSet.scala

```diff
@@ -240,7 +240,7 @@ class BitSet(numBits: Int) extends Serializable {
    * Compute bit-wise union with another BitSet and overwrite bits in this BitSet with the result.
    */
   def union(other: BitSet): Unit = {
-    require(this.numWords <= other.numWords)
+    require(this.numWords >= other.numWords)
     var ind = 0
     while( ind < this.numWords ) {
```

Comment:
hmm, so the `union` function here ignores the bits in `other` if it's longer than `this`?

Reply:
Yes. 
If we want prevent occasional bugs, we can change 
 require(this.numWords <= other.numWords)
to 
 require(this.numWords == other.numWords)?

Reply:
yea, that looks better

Reply:
All right, will send a PR for that.

Reply:
Force equal length of two bitsets sounds fine. btw Java `java.util.BitSet` allows two bitsets to be different lengths (`or()`, `and()`, etc).